include/swift/AST/AutoDiff.h

```diff
@@ -316,6 +316,14 @@ struct SILAutoDiffIndices {
                [&s](unsigned p) { s << p; }, [&s]{ s << ' '; });
     s << "))";
   }
+
+  std::string mangle() const {
```

Comment:
Can you define a new AutoDiffMangler or similar, even if its still doing stuff like this? It will be easier to move to a real mangling in the future (which I encourage you to do).

Reply:
Kind of following the example in ASTMangler? I haven't read through that and understood how it works yet, but I'll do that and then implement something like it, in a later PR.